cocos/physics/CCPhysicsJoint.cpp

```diff
@@ -34,6 +34,83 @@
 
 NS_CC_BEGIN
 
+template<typename T>
+class Optional {
+
+public:
+    Optional() {}
+    Optional(T d) :_isSet(true), _data(d) {}
+    Optional(const Optional &t) : _isSet(t._isSet), _data(t._data) {}
+
+    //bool isNull()       const { return !_isSet; }
+    //bool isDefineded()  const { return _isSet; }
+    //bool isEmpty()      const { return !_isSet; }
+
+    T    get()          const { CCASSERT(_isSet, "data should be set!"); return _data; }
+    void set(T d)             { _isSet = true; _data = d; }
+
+private:
+    bool _isSet = false;
+    T _data;
+};
+
+class WriteCache {
+public:
+    Optional<Vec2>  _grooveA;
+    Optional<Vec2>  _grooveB;
+    Optional<Vec2>  _anchr1;
+    Optional<Vec2>  _anchr2;
+    Optional<float> _min;
+    Optional<float> _max;
+    Optional<float> _distance;
+    Optional<float> _restLength;
+    Optional<float> _restAngle;
+    Optional<float> _stiffness;
+    Optional<float> _damping;
+    Optional<float> _angle;
+    Optional<float> _phase;
+    Optional<float> _ratchet;
+    Optional<float> _ratio;
+    Optional<float> _rate;
+};
+
+#if (defined(__GNUC__) && __GNUC__ >= 4) || defined(__clang__)
+#define LIKELY(x)   (__builtin_expect((x), 1))
+#define UNLIKELY(x) (__builtin_expect((x), 0))
+#else
+#define LIKELY(x)   (x)
+#define UNLIKELY(x) (x)
+#endif
+
+
+#define CC_PJOINT_CACHE_READ(field)         \
+    do {                                    \
+    if(UNLIKELY(_initDirty))                \
```

Comment:
_initDirty is initialized to `true` at constructor, why it is unlikely?

Reply:
It will remain false after `initJoint()` is invoked. And I think it should be predicted to be `false` for the most time.